packages/vm/src/runBlock.ts

```diff
@@ -42,17 +43,35 @@ const parentBeaconBlockRootAddress = Address.fromString(
   '0xbEac00dDB15f3B6d645C48263dC93862413A222D'
 )
 
+let enableProfiler = false
+let stateRootCPLabel = ''
+let processTxsLabel = ''
+let withdrawalsRewardsCommitLabel = ''
+
 /**
  * @ignore
  */
 export async function runBlock(this: VM, opts: RunBlockOpts): Promise<RunBlockResult> {
+  if (this._opts.profilerOpts?.reportAfterBlock === true) {
+    enableProfiler = true
+  }
+
   const state = this.stateManager
   const { root } = opts
   const clearCache = opts.clearCache ?? true
   const setHardfork = opts.setHardfork ?? false
   let { block } = opts
   const generateFields = opts.generate === true
 
+  if (enableProfiler) {
+    const title = `Profiler run - Block ${block.header.number} (${bytesToHex(block.hash())} with ${
+      block.transactions.length
+    } txs`
+    console.log(title)
+    stateRootCPLabel = 'New state root, DAO HF, checkpoints, block validation'
```

Comment:
Can we also time the entire `runBlock` method? So it can be seen what relatively takes the most time.